Populate LibraryRun upon SequenceRun PendingAnalysis

**Context:**

- Creation of LibraryRun is too early such that SampleSheet.csv file has not yet avail. It is good enough to populate upon SequenceRun status PendingAnalysis.
- See CloudWatch logs:

LogGroup:
```
/aws/lambda/data-portal-api-prod-sqs_iap_event_processor
```

LogStreams:
```
2021/11/22/[$LATEST]f62bb729a5894d19b966db9fbefaa3cf
2021/11/22/[$LATEST]49ac2e79bfa24176afe10df3d6b41c7a
```

Exception:

```
WARNING  Please specify a single file. Found 0 files at gds://bssh.<masked>/Runs/211122_A01052_0065_<masked>_r.RJ<masked>-pzAA/SampleSheet.csv
...
[ERROR] ValueError: Can not download sample sheet from GDS: gds://bssh.<masked>/Runs/211122_A01052_0065_<masked>_r.RJ<masked>-pzAA/SampleSheet.csv
Traceback (most recent call last):
  File "/var/task/data_processors/pipeline/lambdas/sqs_iap_event.py", line 55, in handler
    handle_bssh_run_event(message, event_action, event_type, context)
  File "/var/task/data_processors/pipeline/lambdas/sqs_iap_event.py", line 82, in handle_bssh_run_event
    library_run_list: List[LibraryRun] = library_run_srv.create_library_run_from_sequence({
  File "/var/lang/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/var/task/data_processors/pipeline/services/library_run_srv.py", line 38, in create_library_run_from_sequence
    samplesheet_json = liborca.get_samplesheet_to_json(gds_volume=gds_volume_name, samplesheet_path=samplesheet_path)
  File "/var/task/data_processors/pipeline/tools/liborca.py", line 170, in get_samplesheet_to_json
    return get_samplesheet(gds_volume, samplesheet_path).to_json()
  File "/var/task/data_processors/pipeline/tools/liborca.py", line 161, in get_samplesheet
    raise ValueError(reason)
```


**Actions:**

- Just populate upon PendingAnalysis should be enough

https://github.com/umccr/data-portal-apis/blob/e8206e803d2767fe646f95f324b7666c5c34c641/data_processors/pipeline/lambdas/sqs_iap_event.py#L83-L90

--- a/data_processors/pipeline/lambdas/sqs_iap_event.py
+++ b/data_processors/pipeline/lambdas/sqs_iap_event.py
@@ -15,7 +15,6 @@ from typing import List
 import logging
 
 from data_portal.models.sequencerun import SequenceRun
-from data_portal.models.sequence import SequenceStatus
 from data_portal.models.libraryrun import LibraryRun
 from data_processors.pipeline.services import sequence_run_srv, notification_srv, sequence_srv, library_run_srv
 from data_processors.pipeline.lambdas import bcl_convert, orchestrator
@@ -80,7 +79,10 @@ def handle_bssh_run_event(message, event_action, event_type, context):
         # Create or update Sequence record from BSSH Run event payload
         seq = sequence_srv.create_or_update_sequence_from_bssh_event(payload)
 
-        if seq.status == SequenceStatus.STARTED.value or sqr.status.lower() == "PendingAnalysis".lower():
+        # Using bssh.runs event status PendingAnalysis as trigger for next event
+        # See https://github.com/umccr-illumina/stratus/issues/95
+        if sqr.status.lower() == "PendingAnalysis".lower():
+            # populate LibraryRun records
             library_run_list: List[LibraryRun] = library_run_srv.create_library_run_from_sequence({
                 'instrument_run_id': seq.instrument_run_id,
                 'run_id': seq.run_id,
@@ -89,9 +91,7 @@ def handle_bssh_run_event(message, event_action, event_type, context):
                 'sample_sheet_name': seq.sample_sheet_name,
             })
 
-        # Once Sequence Run status is good, launch bcl convert workflow
-        # Using bssh.runs event status PendingAnalysis for now, See https://github.com/umccr-illumina/stratus/issues/95
-        if sqr.status.lower() == "PendingAnalysis".lower():
+            # launch bcl convert workflow
             bcl_convert.handler({
                 'gds_volume_name': sqr.gds_volume_name,
                 'gds_folder_path': sqr.gds_folder_path,

